ui/components/app/assets/nfts/nfts-items/nfts-items.js

```diff
@@ -116,39 +118,40 @@ export default function NftsItems({
     dispatch,
   ]);
 
-  const history = useHistory();
+  const getAssetImageUrlAndUpdate = async (image, nft) => {
+    const nftImage = await getAssetImageURL(image, ipfsGateway);
+    const updatedNFt = {
+      ...nft,
+      ipfsImageUpdated: nftImage,
+    };
+    return updatedNFt;
+  };
 
-  const renderCollectionImage = (collectionImage, collectionName) => {
-    if (collectionImage?.startsWith('ipfs') && !ipfsGateway) {
-      return (
-        <div className="nfts-items__collection-image-alt">
-          {collectionName?.[0]?.toUpperCase() ?? null}
-        </div>
-      );
-    }
-    if (!openSeaEnabled && !collectionImage?.startsWith('ipfs')) {
-      return (
-        <div className="nfts-items__collection-image-alt">
-          {collectionName?.[0]?.toUpperCase() ?? null}
-        </div>
-      );
-    }
+  useEffect(() => {
```

Comment:
Could you describe the thinking behind this useEffect?  Wasn't clear to me why it's needed now vs before.

Reply:
Those changes are related to the `async getFormattedIpfsUrl `update; In a lot of our components the `src` is computed directly using   `src={getAssetImageURL(collectionImage, ipfsGateway)}`; now since `getAssetImageURL(image, ipfsGateway)` is returning a promise, i added the `useEffect` logic to resolve those.

When the logic of useEffect is simple; i use the `useGetAssetImageUrl` hook;